libsoftwarecontainer/unit-test/iptableentry_unittest.cpp

```diff
@@ -0,0 +1,100 @@
+
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+
+#include "gateway/iptableentry.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+class MockIPTableEntry :
+    public IPTableEntry
+{
+public:
+    MockIPTableEntry():IPTableEntry()
+    {
+        m_type = "INPUT";
+        m_defaultTarget = Target::ACCEPT;
+    }
+};
+
+class IPTableEntryTest : public ::testing::Test
+{
+protected:
+    ::testing::NiceMock<MockIPTableEntry> ipTable;
+};
+
+//test policies
+TEST_F(IPTableEntryTest, TestIPTableEntryPolicies) {
+    IPTableEntry::Rule r;
+    r.host = "127.0.0.1/16";
+    r.ports.any = true;
+    r.ports.multiport = true;
+    r.ports.ports = "80,8080";
+    r.target = IPTableEntry::Target::ACCEPT;
+    ASSERT_EQ("iptables -P INPUT ACCEPT", ipTable.interpretPolicy());
+
+    ipTable.m_defaultTarget = IPTableEntry::Target::DROP;
+    ASSERT_EQ("iptables -P INPUT DROP", ipTable.interpretPolicy());
+
+    //Test wrong default Policy
+    ipTable.m_defaultTarget = IPTableEntry::Target::REJECT;
+    ASSERT_EQ("", ipTable.interpretPolicy());
+}
```

Comment:
Does these tests need to be run in a sequence? Otherwise split them into multiple tests.